bpf/tests/bpf-prog-test-run/bpf-prog-test-run.go

```diff
@@ -0,0 +1,487 @@
+		return fmt.Errorf("test run failed: %v", err)
+	}
+	if bpfRet != 0 { // CT_ACT_OK
+		return errors.New("unexpected return value")
+	}
+	if !reflect.DeepEqual(bufOut, packetSyn.Bytes()) {
+		return errors.New("unexpected data modification")
+	}
+	err = dumpDebugMessages(eventsReader)
+	if err != nil {
+		return fmt.Errorf("dumpDebugMessages failed: %v", err)
+	}
+	bpfCt, err := ct4TcpMapFromBPF(bpfCtMap)
+	if err != nil {
+		return fmt.Errorf("ct4TcpMapFromBPF failed: %v", err)
+	}
+	fmt.Printf("CT Entries:\n")
+	for key, val := range bpfCt {
+		fmt.Printf(" key=%s val=%s", &key, &val)
+	}
+
+	now, err := bpf.GetMtime()
+	if err != nil {
+		return errors.New("GetMtime failed")
+	}
+	// Second Packet: RST
+	prog2 := spec.Programs["test_ct4_rst2"]
+	if prog2 == nil {
+		return errors.New("did not find test_ct4_rst2 program")
+	}
+	bpfRet, bufOut, err = prog2.Test(packetRst.Bytes())
+	if err != nil {
+		return fmt.Errorf("test run failed: %v", err)
+	}
+	err = dumpDebugMessages(eventsReader)
+	if err != nil {
+		return fmt.Errorf("dumpDebugMessages failed: %v", err)
+	}
+	bpfCt, err = ct4TcpMapFromBPF(bpfCtMap)
+	if err != nil {
+		return fmt.Errorf("ct4TcpMapFromBPF failed: %v", err)
+	}
+	fmt.Printf("CT Entries:\n")
+	for key, val := range bpfCt {
+		fmt.Printf(" key=%s val=%s", key.ToHost(), &val)
+	}
+	if bpfRet != 0 { // CT_ACT_OK
+		return errors.New("unexpected return value")
+	}
+	if !reflect.DeepEqual(bufOut, packetRst.Bytes()) {
+		return errors.New("unexpected data modification")
+	}
+
+	key := ctmap.CtKey4Global{
+		TupleKey4Global: tuple.TupleKey4Global{
+			TupleKey4: tuple.TupleKey4{
+				SourceAddr: types.IPv4{10, 3, 0, 20},
+				DestAddr:   types.IPv4{10, 3, 0, 10},
+				SourcePort: byteorder.HostToNetwork(uint16(3010)).(uint16),
+				DestPort:   byteorder.HostToNetwork(uint16(3020)).(uint16),
+				NextHeader: 6,
+				Flags:      0,
+			},
+		},
+	}
+	val := bpfCt[key]
+	expires := (val.Lifetime - uint32(now/1000000000))
+	fmt.Printf("Entry expires in %ds\n", expires)
+
+	return nil
+}
+
+func main() {
+
+	lim := unix.Rlimit{
+		Cur: unix.RLIM_INFINITY,
+		Max: unix.RLIM_INFINITY,
+	}
+	err := unix.Setrlimit(unix.RLIMIT_MEMLOCK, &lim)
+	if err != nil {
+		log.Fatalf("setrlimit: %v", err)
+	}
+
+	fname := "bpf_ct_tests.o"
+	fnamePatched := "bpf_ct_tests_patched.o"
+	err = patchElf(fname, fnamePatched)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	coll, err := ebpf.LoadCollection(fnamePatched)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"filename": fnamePatched,
+		}).Fatalf("failed to load: %s\n", err)
+	}
+
+	err = testNop(coll)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"test":  "nop",
+			"error": err,
+		}).Fatal("test failed")
+	}
+
+	err = testMap(coll)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"test":  "map",
+			"error": err,
+		}).Fatal("test failed")
+	}
+
+	err = testCt4Rst(coll)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"test":  "map",
+			"error": err,
+		}).Fatal("test failed")
+	}
```

Comment:
Hmm, if I understand you mean I could do something like:

```
for fn := range  testFns {
    fn(...)
}
```

I'd rather not, since we might want to pass different arguments to the test functions eventually.